storage/s3/src/main/java/io/aiven/kafka/tieredstorage/storage/s3/MetricCollector.java

```diff
@@ -76,35 +68,32 @@ private Sensor createSensor(final String name) {
     }
 
     @Override
-    public void collectMetrics(final Request<?> request, final Response<?> response) {
-        final AWSRequestMetrics awsRequestMetrics = request.getAWSRequestMetrics();
-        if (awsRequestMetrics == null) {
-            return;
-        }
-        final List<Object> requestTypes = awsRequestMetrics.getProperty(AWSRequestMetrics.Field.RequestType);
-        if (requestTypes == null || requestTypes.isEmpty()) {
-            return;
-        }
-        final Object requestType = requestTypes.get(0);
-        if (requestType == null) {
-            return;
-        }
-
-        // Try to arrange them by likelihood, at least GetObjectRequest should be first.
-        if (GetObjectRequest.class.getSimpleName().equals(requestType)) {
+    public void publish(final MetricCollection metricCollection) {
+        if (metricCollection.metricValues(CoreMetric.OPERATION_NAME).contains("GetObject")) {
             this.getObjectRequests.record();
-        } else if (UploadPartRequest.class.getSimpleName().equals(requestType)) {
+        }
+        if (metricCollection.metricValues(CoreMetric.OPERATION_NAME).contains("UploadPart")) {
```

Comment:
I think `metricCollection.metricValues(CoreMetric.OPERATION_NAME)` should be in a variable (it does some iteration inside, which maybe not super optimal).